cognite_toolkit/_builtin_modules/contextualization/cdf_entity_matching/functions/contextualization_connection_writer/handler.py

```diff
@@ -0,0 +1,331 @@
+import json
+import sys
+import traceback
+from collections import defaultdict
+from collections.abc import Iterable, Sequence
+from typing import ClassVar, Literal, TypeVar, cast
+
+from cognite.client.config import global_config
+
+# Do not warn the user about feature previews from the Cognite-SDK we use in Toolkit
+# ruff: noqa: E402
+global_config.disable_pypi_version_check = True
+global_config.silence_feature_preview_warnings = True
+import yaml
+from cognite.client import CogniteClient
+from cognite.client import data_modeling as dm
+from cognite.client.data_classes import ExtractionPipelineRunWrite, RowWrite
+from cognite.client.data_classes.data_modeling.cdm.v1 import CogniteDiagramAnnotation, CogniteDiagramAnnotationApply
+from pydantic import BaseModel, model_validator
+from pydantic.alias_generators import to_camel
+
+if sys.version_info >= (3, 11):
+    from typing import Self
+else:
+    from typing_extensions import Self
+
+
+FUNCTION_ID = "connection_writer"
+EXTRACTION_PIPELINE_EXTERNAL_ID = "ctx_files_direct_relation_write"
+EXTERNAL_ID_LIMIT = 256
+EXTRACTION_RUN_MESSAGE_LIMIT = 1000
+
+
+def handle(data: dict, client: CogniteClient) -> dict:
+    try:
+        execute(data, client)
+    except Exception as e:
+        tb = traceback.extract_tb(e.__traceback__)
+        last_entry_this_file = next((entry for entry in reversed(tb) if entry.filename == __file__), None)
+        suffix = ""
+        if last_entry_this_file:
+            suffix = f" in function {last_entry_this_file.name} on line {last_entry_this_file.lineno}: {last_entry_this_file.line}"
+
+        status: Literal["failure", "success"] = "failure"
+        # Truncate the error message to 1000 characters the maximum allowed by the API
+        prefix = f"ERROR {FUNCTION_ID}: "
+        error_msg = f'"{e!s}"'
+        message = prefix + error_msg + suffix
+        if len(message) >= EXTRACTION_RUN_MESSAGE_LIMIT:
+            error_msg = error_msg[: EXTRACTION_RUN_MESSAGE_LIMIT - len(prefix) - len(suffix) - 3 - 1]
+            message = prefix + error_msg + '..."' + suffix
+    else:
+        status = "success"
+        message = FUNCTION_ID
+
+    client.extraction_pipelines.runs.create(
+        ExtractionPipelineRunWrite(extpipe_external_id=EXTRACTION_PIPELINE_EXTERNAL_ID, status=status, message=message)
+    )
```

Comment:
Maybe this deserves a try...except since the Function may have succeeded but fail to report it due to for example missing ACL